### PR TITLE
Re-tune all hyperparameters for v2

### PR DESCRIPTION
**Pull request by MilesCranmer**
## Summary

This PR updates the default hyperparameters used by `Options()` and `defaults >= v"2.0.0-"`. The previous profile remains available through `defaults=v"1.0.0"`.

`niterations` is unchanged. Each iteration performs more evolutionary work, so the default search runs longer.

## Tuning benchmark

The hyperparameters were selected using a fixed suite of exact equations, noisy equations, feature-selection problems, and real regression datasets.

| Task | Target or dataset | Train/test | Operators | `maxsize` | Score |
|---|---|---:|---|---:|---|
| Damped oscillator | `2.3 exp(-t / 3.0) cos(1.7t + 0.4)` with Gaussian noise `σ = 0.05 std(y)` | 4096/4096 | `+ - * / exp cos` | 30 | Normalized Pareto volume |
| Smooth Friedman equation | `sin(π x₀ x₁) + (x₂ - 0.5)²` | 4096/4096 | `+ - * / sin` | 30 | Soft exact recovery, true complexity 13 |
| Feature-selection stress test | `sin(x₀) + x₁x₂`, with `x₃,…,x₉` as irrelevant inputs | 4096/4096 | `+ - * / sin` | 50 | Soft exact recovery, true complexity 7 |
| Log/square-root interaction | `log(x₀) / sqrt(x₁ + x₂)` | 4096/4096 | `+ - * / log sqrt` | 30 | Soft exact recovery, true complexity 9 |
| Noisy Friedman-1 | `10 sin(π x₀ x₁) + 20(x₂ - 0.5)² + 10x₃ + 5x₄`, with `x₅,…,x₉` irrelevant and Gaussian noise `σ = 0.1 std(y)` | 4096/4096 | `+ - * / sin` | 50 | Normalized Pareto volume |
| Dense linear regression | `Σᵢ wᵢxᵢ`, with 30 relevant features, fixed `wᵢ ~ N(0,1)`, and Gaussian noise `σ = 0.05 std(y)` | 4096/4096 | `+ - * /` | 100 | Normalized Pareto volume |
| California housing | Standardized 10,000-example subset of California housing | 8000/2000 | `+ - * / sin cos exp log sqrt tanh` | 50 | Normalized log-MSE improvement |
| ESL | Standardized PMLB `1027_ESL` | 390/98 | `+ - * / sin cos exp log sqrt tanh` | 30 | Normalized log-MSE improvement |
| PM10 | Standardized PMLB `522_pm10` | 400/100 | `+ - * / sin cos exp log sqrt tanh` | 50 | Normalized log-MSE improvement |
| Galaxy | Standardized PMLB `690_visualizing_galaxy` | 258/65 | `+ - * / sin cos exp log sqrt tanh` | 30 | Normalized log-MSE improvement |

### Scoring

Each fit was evaluated on its test set. The scoring rule depended on the task.

For noisy synthetic tasks, the score was the normalized Pareto volume of test MSE against expression complexity. The Pareto frontier was constructed after keeping the lowest test MSE at each complexity. Its convex-hull area was computed in `(log10(complexity), log10(MSE))` space and normalized by the rectangle spanning complexity `1` to `maxsize + 1` and MSE from the theoretical noise floor `σ²` to the mean-prediction baseline.

For exact tasks, every member `i` of the discovered Pareto frontier received

```text
fit_i        = 1 / (1 + MSE_i / 1e-10)
complexity_i = exp(-(log(2) / 4) max(0, c_i - c_true))
score        = max_i fit_i complexity_i
```

This gives a score near one for an exact low-complexity recovery, while smoothly penalizing residual error and expressions more complex than the known target.

For real datasets, the score used the best test-MSE member of the discovered Pareto frontier:

```text
score = clip(
    (log(MSE_baseline) - log(MSE_model))
    / (log(MSE_baseline) - log(MSE_oracle)),
    0,
    1,
)
```

`MSE_baseline` is the test MSE of the training-set mean. `MSE_oracle` is the test MSE of an XGBoost model trained on the same standardized split.

Each configuration was evaluated with three repetitions per task and a five-minute limit per fit. The task score was the mean over its three repetitions, and the global score was the unweighted mean of the ten task scores.

### Tuning procedure

The tuning was conducted in two stages. Earlier distributed Optuna sweeps provided candidate configurations and broad information about the search space. The final selection campaign then froze the ten-task benchmark above and switched to manual, evidence-directed tuning. That stage used deterministic named configurations organized into coarse grids, ablations, local neighborhoods, and recombinations of strong configurations. After each wave, the next configurations were chosen from the per-task scores, weakest tasks, family-level results, and failure modes observed in completed trials. Optuna did not generate configurations during this final manual stage.

The manual campaign evaluated 5,180 complete unique configurations on the full benchmark. The best audited configuration scored `0.8266`; the campaign stopping threshold was `0.90`.

The values in this PR use that configuration and its surrounding high-performing families as a starting point. Several values were rounded or made more conservative where the campaign did not support the raw extreme value reliably.

## Search and selection defaults

| Hyperparameter | Previous | V2 |
|---|---:|---:|
| `populations` | 31 | 18 |
| `population_size` | 27 | 21 |
| `ncycles_per_iteration` | 380 | 560 |
| `adaptive_parsimony_scaling` | 1040 | 20 |
| `crossover_probability` | 0.0259 | 0.20 |
| `alpha` | 3.17 | 0.1125 |
| `perturbation_factor` | 0.129 | 0.02 |
| `use_frequency` | `true` | `false` |
| `use_frequency_in_tournament` | `true` | `true` |
| `optimizer_nrestarts` | 2 | 3 |
| `optimizer_probability` | 0.14 | 0.28 |
| `optimizer_iterations` | 8 | 10 |
| `optimizer_f_calls_limit` | 10,000 | 3,000 |
| `tournament_selection_n` | 15 | 10 |
| `tournament_selection_p` | 0.982 | 0.74 |
| `fraction_replaced_hof` | 0.0614 | 0.5 |

The initial v2 profile disabled frequency-based adaptive parsimony during tournament selection, which caused the custom-types example to fail. Further investigation showed that this mechanism still materially helps some search problems: without it, the custom-types example performed substantially worse. The v2 defaults therefore keep `use_frequency_in_tournament=true` while leaving `use_frequency=false`.

## Mutation weights

Mutation weights are relative and normalized during mutation selection.

| Mutation | Previous | V2 |
|---|---:|---:|
| `mutate_constant` | 0.0353 | 0.0346 |
| `mutate_operator` | 3.63 | 0.01 |
| `mutate_feature` | 0.10 | 0.15 |
| `swap_operands` | 0.00608 | 0.80 |
| `rotate_tree` | 1.42 | 0.20 |
| `add_node` | 0.0771 | 4.70 |
| `insert_node` | 2.44 | 0.06 |
| `delete_node` | 0.369 | 2.10 |
| `simplify` | 0.00148 | 0.0020 |
| `randomize` | 0.00695 | 0.0001 |
| `do_nothing` | 0.431 | 0.26 |

## Runtime impact

Representative `example.jl` timings:

| Mode | Previous | V2 |
|---|---:|---:|
| Multithreaded | 15.3 s | 35.7 s |
| Serial | 33.2 s | 78.1 s |

The longer runtime is intentional because many users rely on the defaults without adjusting the search budget. Users who prefer a smaller budget can reduce `niterations`.

### CODE REVIEW

**Comment by MilesCranmer** [comment]
I'm just not really seeing consistent improvements... So to be safe I'm going to close this and go with the simpler change of #643.
